moshi/src/main/java/com/squareup/moshi/JsonWriter.java

```diff
@@ -180,7 +181,11 @@
   /**
    * Creates a new instance that writes a JSON-encoded stream to {@code sink}.
    */
-  public JsonWriter(BufferedSink sink) {
+  public JsonWriter(Sink sink) {
```

Comment:
I can add Javadoc to that effect if you think it's likely to be a point of user confusion / problems.